wgpu-hal/src/gles/egl.rs

```diff
@@ -757,13 +756,30 @@ impl crate::Surface<super::Api> for Surface {
             );
         }
 
-        let format_desc = device.shared.describe_texture_format(config.format);
         let gl = &device.shared.context;
+
+        // Check renderer version to see if we need to workaround intel mesa bug
+        if self.enable_intel_mesa_fastclear_workaround.is_none() {
+            let renderer = gl.get_parameter_string(glow::RENDERER);
+
+            // Check renderer string for buggy cards
+            self.enable_intel_mesa_fastclear_workaround = Some(
+                renderer.contains("Mesa")
+                    && renderer.contains("Intel")
+                    && renderer.contains("CML GT2"),
```

Comment:
This is going to be way to restrictive. This limits it to Comet Lake and only the GT2 model of that (which is the mid-range gpu in the lineup). https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/4972/diffs?diff_id=75888#22f5d1004713c9bbf857988c7efb81631ab88f99_323_327 comment seems to suggest the issue applies to all skylake based machines which matches with the issue not showing up on my haswell machine.

Here are the PCI ids for all intel cards: https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/include/pci_ids/i965_pci_ids.h.

What we need to do is match on all idents which are "**L". You could write something like:

`renderer.split().any(|substr| substr.len() == 3 && substr[2] == 'L')` as the model check. You should also point back to this documentation to show why we're making these decisions.

Reply:
OK, great, I was wondering what exactly those model numbers meant. :)